cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -0,0 +1,376 @@
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "request doest not contain any item to remove",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
+	}
+
+	if len(remove) > 0 {
+		if err := removeManagedNamespaces(context.Background(), client, remove, projectID, clusterName); err != nil {
+			_, _ = logger.Debugln("Error -Unable to remove Namespace")
+		}
+	}
+	if currentModel.RemoveAllZoneMapping != nil && *currentModel.RemoveAllZoneMapping {
+		_, _, err := client.GlobalClusters.DeleteCustomZoneMappings(context.Background(), projectID, clusterName)
+		if err != nil {
+			return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to custom zones : %s", err.Error()),
+				cloudformation.HandlerErrorCodeInvalidRequest), nil
+		}
+	}
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete Complete",
+	}, nil
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-global-cluster-config")
+}
+
+func isExist(client mongodbatlas.Client, currentModel *Model) bool {
+	setup()
+	config, _, err := ReadConfig(client, currentModel)
+	_, _ = logger.Debugf("%v", config)
+	if err == nil && config != nil {
+		return true
+	}
+	return false
+}
+
+// function to validate inputs to all actions
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
+
+func removeManagedNamespaces(ctx context.Context, conn *mongodbatlas.Client, remove []ManagedNamespace, projectID, clusterName string) error {
+	for _, m := range remove {
+		addManagedNamespace := &mongodbatlas.ManagedNamespace{
+			Collection:     cast.ToString(m.Collection),
+			Db:             cast.ToString(m.Db),
+			CustomShardKey: cast.ToString(m.CustomShardKey),
+		}
+		addManagedNamespace.IsCustomShardKeyHashed = pointy.Bool(*m.IsCustomShardKeyHashed)
+		addManagedNamespace.IsShardKeyUnique = pointy.Bool(*m.IsShardKeyUnique)
+		_, _, err := conn.GlobalClusters.DeleteManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+
+		if err != nil {
+			_, _ = logger.Warnf("error while removing namespace:%+v", err)
+		}
+	}
+
+	return nil
+}
+
+func addManagedNamespaces(ctx context.Context, client *mongodbatlas.Client, nameSpaces []ManagedNamespace, projectID, clusterName string) error {
+	if len(nameSpaces) == 0 {
+		return nil
+	}
+	for _, mn := range nameSpaces {
+		addManagedNamespace := &mongodbatlas.ManagedNamespace{
+			Collection:     cast.ToString(mn.Collection),
+			Db:             cast.ToString(mn.Db),
+			CustomShardKey: cast.ToString(mn.CustomShardKey),
+		}
+		addManagedNamespace.IsCustomShardKeyHashed = mn.IsCustomShardKeyHashed
+		addManagedNamespace.IsShardKeyUnique = mn.IsShardKeyUnique
+		_, _, err := client.GlobalClusters.AddManagedNamespace(ctx, projectID, clusterName, addManagedNamespace)
+		if err != nil {
+			_, _ = logger.Warnf("error while adding namespace:%+v", err)
+		}
+	}
+
+	return nil
```

Comment:
here is the same, there is no return err, we are always return nil, do we need to return something then?

Reply:
No.consider scenario we are adding 5 name spaces and one is already exist .So we need log the error for that case alone and add remaining